### ISSUE TITLE
Question about weight_eos

### ISSUE THREAD

**Issue by violet-sto** [DreamLM/DreamOn#3]
Hi,

Thanks for the excellent work. I notice that you downweight the loss on the <eos> token. I would like to know whether this approach is used in the pre-training process of Dream8B or Dreamcoder?

**Comment by WilliamZR** [comment]
Hi, 

Downweighting loss of EOS/PAD is only used in this work.  

We use a context-adaptive token-level noise rescheduling during pretraining of Dream-7B. As for DreamCoder, we truncate responses to avoid over-padded training targets.

**Comment by violet-sto** [comment]
Thanks for the reply!

Currently, the code for Dream-7B has not been open-sourced. Is it possible to share the implementation details of context-adaptive token-level noise rescheduling?

**Comment by WilliamZR** [comment]
I think the release is still in progess... As for now, please checkout Equation 5 in our [technical report](https://arxiv.org/pdf/2508.15487) on Arxiv for more details.

**Comment by WilliamZR** [comment]
The training code for Dream-7B is now available at https://github.com/DreamLM/Dream.
